stonesoup/updater/kalman.py

```diff
@@ -562,3 +558,88 @@ def _posterior_covariance(self, hypothesis):
+
+    where,
+
+    .. math::
+
+        H_i &= h^{\prime}(x_{k,i}), \ K_i = P_{k|k-1} H_i^T (H_i P_{k|k-1} H_i^T + R)^{-1}
+
+    and
+
+    .. math::
+
+        x_{k,0} &= x_{k|k-1} \ \mathrm{and} \ P_{k,0} = P_{k|k-1}
+
+    It inherits from the ExtendedKalmanUpdater as it uses the same linearisation of the sensor
+    function via the :meth:`_measurement_matrix()` function.
+    """
+
+    tolerance = Property(float, default=1e-6,
+                         doc="The value of the difference in the measure used as a stopping "
+                             "criterion.")
+    measure = Property(Measure, default=Euclidean(), doc="")
+
+    def update(self, hypothesis, **kwargs):
+        r"""The iterated Kalman update method. Given a hypothesised association between a predicted
+        state or predicted measurement and an actual measurement,
+        calculate the posterior state.
+
+        Parameters
+        ----------
+        hypothesis : :class:`~.SingleHypothesis`
+            the prediction-measurement association hypothesis. This hypothesis
+            may carry a predicted measurement, or a predicted state. In the
+            latter case a predicted measurement will be calculated.
+        **kwargs : various
+            These are passed to the measurement model function
+
+        Returns
+        -------
+        : :class:`~.GaussianStateUpdate`
+            The posterior state Gaussian with mean :math:`\mathbf{x}_{k|k}` and
+            covariance :math:`P_{x|x}`
+
+        """
+
+        # Record the starting point
+        prev_state = hypothesis.prediction
+
+        # Get the measurement model
+        measurement_model = self._check_measurement_model(hypothesis.measurement.measurement_model)
+
+        # The first iteration is just the application of the EKF
+        post_state = super().update(hypothesis, **kwargs)
+
+        # Now update the measurement prediction mean and loop
+        while self.measure(prev_state, post_state) > self.tolerance:
+
+            # These lines effectively bypass the predict_measurement function in update()
+            # by attaching new linearised quantities to the measurement_prediction. Those
+            # would otherwise be calculated (from the original prediction) by the update() method.
+            hh = self._measurement_matrix(post_state, measurement_model=measurement_model)
+
+            post_state.hypothesis.measurement_prediction.state_vector = \
+                measurement_model.function(post_state, noise=None) + \
+                hh@(hypothesis.prediction.state_vector - post_state.state_vector)
+
+            cross_cov = self._measurement_cross_covariance(hypothesis.prediction, hh)
+            post_state.hypothesis.measurement_prediction.cross_covar = cross_cov
+            post_state.hypothesis.measurement_prediction.covar = \
+                self._innovation_covariance(cross_cov, hh, measurement_model)
+
+            prev_state = post_state  # Does this require a copy?
```

Comment:
So this would require a copy if you don't want to modify the original measurement_prediction that was on the hypothesis. Not sure if this is really an issue to be honest.

If you did want to copy, suggest you just do it once before the while loop, after the first call to `super`. You shouldn't need a _deepcopy_, other than also copying `measurement_prediction` because the way you are modifying things.
```python
post_state.hypothesis = copy.copy(post_state.hypothesis)
post_state.hypothesis.measurement_prediction = copy.copy(post_state.hypothesis.measurement_prediction)
```